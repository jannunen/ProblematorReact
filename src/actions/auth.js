import { TRY_AUTH, AUTH_SET_TOKEN, AUTH_REMOVE_TOKEN } from "./actionTypes";



export const authStoreToken = (token) => ({
  type : 'GET_AUTH_SAGA',
  payload : token
  /*
    return dispatch => {
      dispatch(authSetToken(token));
      AsyncStorage.setItem("problemator:auth:token", token);
    };
    */
  });

export const authSetToken = (token) => ({
        type: 'GET_AUTH_SAGA',
        payload : token,
});


export const authGetToken = () => {
    return (dispatch, getState) => {
      const promise = new Promise((resolve, reject) => {
        const token = getState().auth.token;
        if (!token) {
          let fetchedToken;
          AsyncStorage.getItem("problemator:auth:token")
            .catch(err => reject())
            .then(tokenFromStorage => {
              fetchedToken = tokenFromStorage;
              if (!tokenFromStorage) {
                reject();
                return;
              }
            })
            .catch(err => reject());
        } else {
          resolve(token);
        }
      });
      return promise
        .then(token => {
          if (!token) {
            throw new Error("No token found");
          } else {
            return token;
          }
        });
    };
  };
  
  

export const authLogout = () => {
    return dispatch => {
      dispatch(authClearStorage()).then(() => {
        App();
      });
      dispatch(authRemoveToken());
    };
  };
  
  export const authRemoveToken = () => {
    return {
      type: AUTH_REMOVE_TOKEN
    };
  };