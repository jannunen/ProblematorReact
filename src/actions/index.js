export { getProblem, getProblems, selectGym } from './problems';
export { authGetToken, authStoreToken } from './auth';