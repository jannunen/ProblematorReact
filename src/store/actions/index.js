export { addPlace, deletePlace } from './places'
export { getProblems, selectGym } from './problems';
export { authGetToken, authStoreToken } from './auth';