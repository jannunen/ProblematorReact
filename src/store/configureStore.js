import { createStore , combineReducers, compose, applyMiddleware } from 'redux';

import problemsReducer from './reducers/problems'
import authReducer from './reducers/auth'
import groupsReducer from './reducers/groups'
import thunk from 'redux-thunk';

import createSagaMiddleware from 'redux-saga'
import rootSaga from '../sagas/index';

const rootReducer = combineReducers({
    problems : problemsReducer,
    auth : authReducer,
    groups : groupsReducer
});

let composeEnhancers = compose;

if (__DEV__) {
    composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}


const configureStore = () => {
    const sagaMiddleware = createSagaMiddleware();
    //return createStore(rootReducer, composeEnhancers(applyMiddleware(sagaMiddleware)));
    const middleWares = [thunk, sagaMiddleware];
    store  = createStore(rootReducer, composeEnhancers(applyMiddleware(...middleWares)));
    sagaMiddleware.run(rootSaga);
    const action = type => store.dispatch({type})
    return store;
}

export default configureStore;