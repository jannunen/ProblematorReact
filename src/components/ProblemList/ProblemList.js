import * as React  from 'react';
import {
  View,
  Text,
  StyleSheet,
  SectionList,
  TouchableOpacity
} from 'react-native'
import  Icon  from 'react-native-vector-icons/Ionicons';
import { connect } from 'react-redux';
import { getProblems } from '../../store/actions/index';
import { red } from 'ansi-colors';

export class ProblemList extends React.Component {

    componentDidMount = () => {
        this.props.onLoadProblems();
    }

  transformProblemsToSections = () => {
    let sections = [];
    if (this.props.problems) {
      for (var idx in this.props.problems) {
        var item = this.props.problems[idx];
        sections.push({ title: idx, data: item.problems.map((item) => { return ({ ...item }) }) });
      };
    }
    if (sections == null) {
      sections = [({ title : 'Loading...', data : ['Loading']})];
    }
    return sections;
  }

  handleItemClicked = (item) => {
      this.props.handleItemClicked(item);
  }

  listItem = (item, index, section) => {
    return  (
      <TouchableOpacity onPress={() => this.handleItemClicked(item)}>
        <View key={"lic" + item.problemid} style={styles.listItemContainer}>
          <Text key={"grade" + item.problemid} style={styles.listItemLeft}>{item.gradename}</Text>
          <Text key={"color" + item.problemid} style={[styles.listItemColor, { backgroundColor: item.htmlcode }]}> </Text>
          <Text key={"tag" + item.problemid} style={[styles.listItemTag]}>{item.tagshort}</Text>
          <View key={"ric " + item.problemid} style={styles.routeInfoContainer}>
            <Text key={"text" + item.problemid} style={styles.listItemText} key={index}>
              {item.startdefinition} | {item.enddefinition}
            </Text>
            <Text key={"author" + item.problemid} style={[styles.listItemText, styles.listItemRoutesetter]} >
              by {item.author} {item.addedrelative}
            </Text>
          </View>
          <Text key={"stars" + item.problemid} style={styles.listItemStars}>
            <Icon style={styles.likeIcon} name="ios-thumbs-up" size={20} color="#decc00" /><Text style={styles.likes}>{item.c_like}</Text>
            <Icon style={styles.likeIcon} name="ios-heart" size={20} color="red" /> <Text style={styles.likes}>{item.c_love}</Text>
          </Text>
          <Text key={"arrowright" + item.problemid} style={styles.listItemRight}>
            <Icon name="ios-arrow-forward" size={30} color="#decc00" />
          </Text>
        </View>
      </TouchableOpacity>
      );
  }

    render() {
        return (
          <View style={{flex : 1}}>
          {this.props.error != null ? <Text style={styles.errorMessage}>Error: {this.props.error}</Text> : null}
            <SectionList 
            style={styles.problemList}
            updateCellsBatchingPeriod={50}   
            maxToRenderPerBatch={100}
            renderItem={({item, index, section}) => 
                this.listItem (item, index, section)
            } 
            renderSectionHeader={({section: {title}}) => (
                <Text style={styles.sectionHeader}>{title}</Text>
            )}
            keyExtractor={(item, index) => {
                const key= item['problemid']+"_"+index;
                return key;
            }}
            sections={this.transformProblemsToSections()}
            />
          </View>
        )
    }

}


const styles= StyleSheet.create({
  problemList: {
    backgroundColor: "#30312e",
  },
  likeIcon: {
    marginTop : 2
  },
  likes: {
    color : "white",
    marginRight : 3,
    paddingRight : 3,
  },
  listItemLeft : {
    fontSize : 23,
    alignItems : 'center',
    fontWeight : 'bold',
    paddingTop : 6,
    paddingLeft : 2,
    paddingRight : 6,
    color : 'white',
    width : 50,
  },
  listItemColor: {
    marginTop : 11,
    marginLeft : 4,
    width : 15,
    height : 15,
    borderRadius : 5,
    borderWidth : 1,
    borderColor : 'white',
    overflow : 'hidden'
  },
  listItemRight: {
    alignSelf : 'flex-end',
    paddingRight : 8
  },
  listItemTag: {
    fontWeight : 'normal',
    fontSize : 20,
    color : 'white',
    marginTop : 6,
    marginLeft : 2
  },
  listItemRoutesetter: {
    color : '#c0c0c0',
  },
  listItemText : {
    paddingTop : 2,
    paddingLeft : 4,
    color : 'white',
  },
  sectionHeader: {
   height : 34,
    backgroundColor : "#eee",
    fontSize : 20,
    paddingLeft : 8,
    paddingTop : 4,
  },
  routeInfoContainer: {
    alignItems : 'stretch',
    flexDirection : 'column',
    flex : 1

  },
  listItemStars: {
    color : '#decc00',
    fontWeight : 'bold',
    fontSize : 18 ,
    marginTop : 8,
    marginRight : 8
  },
  listItemContainer: {
    flex : 1,
    flexDirection : 'row',
    width : "100%",
    borderColor : 'black',
    borderBottomWidth : 1,
    alignItems : 'stretch',
    height : 40,
  },
  errorMessage: {
    color : 'red',
    fontSize : 18,
    padding : 4,
  }
});


const mapStateToProps = (state) => {
  return {
    problems : state.problems.problems,
    loading : state.problems.loading,
    error : state.problems.error
  }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onLoadProblems: () => dispatch(getProblems()),
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProblemList);