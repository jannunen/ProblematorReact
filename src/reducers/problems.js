import { DEL_BETAVIDEO_PUT, ADD_BETAVIDEO_PUT, GLOBAL_ASCENTS_PUT, DELETE_TICK_PUT,  GET_PROBLEM_PUT, SET_PROBLEMS, PROBLEMS_LOAD_ERROR, PROBLEMS_START_LOADING, SELECT_GYM } from '../actions/actionTypes'

export const initialState = {
    problems: [],
    probleminfos: [],
    globalAscents : [],
    loading : false,
    error : null,
    uiState : 'ready'
}
const reducer = (state = initialState, action) => {
    const payload = action.payload;
    let source = null;
    console.log(action);
    if (payload != null && payload.source != null) {
        source = payload.source;
        delete action.payload.source;
    }
    console.log(action.type,"payload to reducer",payload);
    let newState = null;
    switch (action.type) {
        case 'ALERT_MESSAGE':
        return {
            ...state
            
        }
        break;
        case 'SAVE_OPINION_PUT':
         return {
             ...state,
             probleminfos : {
                 ...state.probleminfos,
                 [source.problemid] : {
                     ...state.probleminfos[source.problemid],
                     c_like :  payload.opinions.likes,
                     c_love :   payload.opinions.loves,
                     c_dislike :  payload.opinions.dislikes,
                 }
             },
             uiState : 'ready'
         }
        break;

        case 'SAVE_FEEDBACK_PUT':
         return {
             ...state,
             uiState : 'ready'
         }
        break;
        case 'UI_LOADING':
         return  {
             ...state,
             uiState : 'loading',
         }
        break;
        case 'UI_STOP_LOADING':
         return  {
             ...state,
             uiState : 'ready',
         }
        break;
        case 'SAVE_TICK_PUT':
         return {
             ...state,
             probleminfos : {
                 ...state.probleminfos,
                 [source.problemid] : {
                     ...state.probleminfos[source.problemid],
                    myticklist :  {
                        ...state.probleminfos[source.problemid].myticklist,
                        [payload.tick.id] : payload.tick 
                    },
                    mytickcount : parseInt(state.probleminfos[source.problemid].mytickcount)+1,
                    ascentcount : parseInt(state.probleminfos[source.problemid].ascentcount)+1,
                }
             },
             uiState : 'ready'
         }
        break;
        case DEL_BETAVIDEO_PUT:
        return  { 
            ...state,
            probleminfos : {
                ...state.probleminfos,
                [payload.problemid] : {
                    ...state.probleminfos[payload.problemid],
                    "betavideos" : 
                        state.probleminfos[payload.problemid].betavideos.filter( item => source.videoid !== item.id),
                    
                }
            },
             uiState : 'ready'
        } 
        break;
        case ADD_BETAVIDEO_PUT:
        return  { 
            ...state,
            probleminfos : {
                ...state.probleminfos,
                [source.problemid] : {
                    ...state.probleminfos[source.problemid],
                    "betavideos" : [
                        ...state.probleminfos[source.problemid].betavideos,
                        action.payload.video
                    ]
                }
            },
             uiState : 'ready'
        } 
        break;
        case GLOBAL_ASCENTS_PUT :
            const pid = payload.problemid;
            return {
                ...state,
                globalAscents : {
                    ...state.globalAscents,
                    [pid] : payload.ascents
                },
                uiState: 'ready'
            }
        break;

        case DELETE_TICK_PUT:
        // That stuff below... I'm not proud of that, but it basically goes 
        // to state.probleminfos[problemid].myticklist and removes the tickid
        // which is being deleted. Thank god for unit tests for these kind of
        // reducers!!!
         return {
             ...state,
             probleminfos : {
                 ...state.probleminfos,
                 [payload.problemid] : {
                     ...state.probleminfos[payload.problemid],
                    myticklist :  Object.keys(state.probleminfos[payload.problemid]['myticklist'])
                    .filter((key) => key !== payload.tickid )
                    .reduce((result,current) => {
                        result[current] =state.probleminfos[payload.problemid]['myticklist'][current];
                        return result;
                    },{})
                 }
             },
             uiState : 'ready'

         }

        break;

        case GET_PROBLEM_PUT:
        // Use 'hashtable' approach to make the searches faster for a certain problem
        console.log("HERE!",action.payload);
         return  {
             ...state,
             probleminfos : {
                 ...state.probleminfos,
                 [action.payload.problem.problemid]: action.payload.problem
             },
             uiState : 'ready'
         }
        break; 

        case PROBLEMS_LOAD_ERROR:
            return {
                ...state,
                uiState: 'ready',
                problems : [],
                error : action.payload.message
            }
        break;
        case SET_PROBLEMS:
            return {
                ...state,
                problems: action.payload,
                uiState: 'ready',
                error : null
            }
            break;

        default:
            console.log("Did not catch the action type!");
            return state;
            break;
    }
}

export default reducer;
