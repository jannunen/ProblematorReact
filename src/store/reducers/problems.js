import { GLOBAL_ASCENTS_PUT, DELETE_TICK_PUT,  GET_PROBLEM_PUT, SET_PROBLEMS, PROBLEMS_LOAD_ERROR, PROBLEMS_START_LOADING, SELECT_GYM } from '../actions/actionTypes'

export const initialState = {
    problems: [],
    probleminfos: [],
    globalAscents : [],
    loading : false,
    error : null,
}
const reducer = (state = initialState, action) => {
    const payload = action.payload;
    switch (action.type) {
        case GLOBAL_ASCENTS_PUT :
            const pid = payload.problemid;
            return {
                ...state,
                globalAscents : {
                    ...state.globalAscents,
                    [pid] : payload.ascents
                }
            }
        break;

        case DELETE_TICK_PUT:
        // That stuff below... I'm not proud of that, but it basically goes 
        // to state.probleminfos[problemid].myticklist and removes the tickid
        // which is being deleted. Thank god for unit tests for these kind of
        // reducers!!!
         return {
             ...state,
             probleminfos : {
                 ...state.probleminfos,
                 [payload.problemid] : {
                     ...state.probleminfos[payload.problemid],
                    myticklist :  Object.keys(state.probleminfos[payload.problemid]['myticklist'])
                    .filter((key) => key !== payload.tickid )
                    .reduce((result,current) => {
                        result[current] =state.probleminfos[payload.problemid]['myticklist'][current];
                        return result;
                    },{})
                 }
             }
         }

        console.log("reducer delete tick",action);
        break;

        case GET_PROBLEM_PUT:
        // Use 'hashtable' approach to make the searches faster for a certain problem
         let newState= {
             ...state,
             probleminfos : {
                 ...state.probleminfos,
                 [action.payload.problem.problemid]: action.payload.problem
             }
         }
         return newState;
        break; 

        case PROBLEMS_START_LOADING:
        return {
            ...state,
            loading : true,
            error : null
        }
        break;
        case PROBLEMS_LOAD_ERROR:
            console.log("reducer",action);
        return {
            ...state,
            loading : false,
            problems : [],
            error : action.payload.message
        }
        break;
        case SET_PROBLEMS:
        console.log("Reducer",action.payload);
            return {
                ...state,
                problems: action.payload,
                loading : false,
                error : null
            }
            break;

        default:
            return state;
            break;
    }
}

export default reducer;
