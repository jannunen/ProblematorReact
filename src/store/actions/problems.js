import { SET_PROBLEMS , PROBLEMS_LOAD_ERROR, PROBLEMS_START_LOADING, SELECT_GYM}  from './actionTypes';
import { API_ENDPOINT} from '../../../config';
import { authGetToken } from './index';
import problems from '../../tests/fixtures/problems';


export const problemsLoadBegin = () => ({
    type : PROBLEMS_START_LOADING
});

export const problemsLoadFailure = (err) => ({
    type : PROBLEMS_LOAD_ERROR,
    error : err
});

export const setProblems = problems => ({
    type : SET_PROBLEMS,
    problems : problems
});


export const getProblems = (gymid) => {
    if (isNaN(gymid)) {
        gymid = 1;
    }
    return (dispatch,getState) => {
        const token = getState().auth.token;
        const url = API_ENDPOINT+"problems/?api-auth-token="+token+"&loc="+gymid;
        console.log("call",url);
        // If developing, just take the problems from a fixture
        dispatch(setProblems(problems));
        return problems;

        return fetch(url)
        .then(res => res.text()) // This is because of JSONP
        .then(textResponse => {
            // Remove first and last, because of JSONP
            console.log("Fixing JSONP");
            if (textResponse.substr(0,1) == "(") {
                console.log("... but only if needed...");
                return JSON.parse(textResponse.substr(1).slice(0, -1));
            } else {
                return JSON.parse(textResponse);
            }
        })
        .then(parsedResp => {
            dispatch(setProblems(parsedResp));
            return parsedResp;
        })
        .catch(err => {
            console.log("Error loading problems",err);
            dispatch(problemsLoadFailure(err));
            return "Error loading problems";
        });
    }
}