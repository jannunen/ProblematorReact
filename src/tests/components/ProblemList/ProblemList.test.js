import React from 'react';
import { shallow } from 'enzyme';
import  ProblemList  from '../../../components/ProblemList/ProblemList';
import problems from '../../fixtures/problems';
import configureStore from 'redux-mock-store';
import { getProblems } from '../../../store/actions/problems';
import thunk from 'redux-thunk';

const middlewares = [thunk]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

let  store,onLoadProblems, history, wrapper;

const initialState = {
    problems : problems,
    loading : false,
    error : null
}

beforeEach(() => {
  });

describe('Testing ProblemList component', () => {

    it('renders as expected', () => {
    wrapper = shallow(
        <ProblemList />,
        { context: { store: mockStore(initialState) } },
      );
      expect(wrapper).toMatchSnapshot();
    })

    it('Fetches problems',async () => {
        fetch.mockResponseOnce(JSON.stringify(problems));
        // Be sure to set the auth to initialstate
        const store = mockStore({auth : { token : 'test_token'}});
        await store.dispatch(getProblems());
        expect(store.getActions()).toMatchSnapshot();
    })
});


/*
test('should all onLoadProblems', () => {
  expect(startAddExpense).toHaveBeenLastCalledWith(expenses[1]);
});
*/
