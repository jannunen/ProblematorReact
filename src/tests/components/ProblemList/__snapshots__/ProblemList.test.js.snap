// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProblemList component Fetches problems 1`] = `
Array [
  Object {
    "problems": Object {
      "A Sl채bi": Object {
        "problems": Array [
          Object {
            "added": "2018-12-27 14:03:16",
            "addedformatted": "27.12.2018",
            "addedrelative": "2 weeks ago",
            "addt": "",
            "attr_bodytension": "0",
            "attr_complex": "1",
            "attr_compression": "0",
            "attr_coordination": "0",
            "attr_crimpers": "0",
            "attr_dyno": "0",
            "attr_endurance": "0",
            "attr_fingery": "0",
            "attr_hooks": "0",
            "attr_jugs": "0",
            "attr_mantle": "1",
            "attr_pinches": "0",
            "attr_pockets": "0",
            "attr_powerful": "1",
            "attr_sensitive": "0",
            "attr_slopers": "0",
            "attr_technical": "0",
            "attr_volumes": "0",
            "australian": "24",
            "author": "Tuukka",
            "c_dangerous": "0",
            "c_dirty": "0",
            "c_dislike": "0",
            "c_like": "0",
            "c_love": "0",
            "chartcolor": "#B370CF",
            "code": "#000",
            "color133": "black",
            "colour": "Black",
            "colourcode": "#000",
            "colourid": "10",
            "crag_route_id": null,
            "enddefinition": "Last hold",
            "ext_id": null,
            "font": "7a",
            "gradeid": "11",
            "gradename": "7A",
            "gymid": "132",
            "hidedate": null,
            "htmlcode": "#000",
            "id": "10",
            "lastWall": "Sl채bi",
            "name": "Black",
            "problemid": "47428",
            "routetype": "boulder",
            "score": "700",
            "slider_complexity": "0",
            "slider_intensity": "0",
            "slider_risk": "0",
            "soontoberemoved": "0",
            "sort": null,
            "south_africa": null,
            "stars": null,
            "startdefinition": "One hold",
            "tag": "BKE2018A034",
            "tagshort": "A034",
            "tapecolour": "#FF0000",
            "textcolor": "#fff",
            "tick": null,
            "uiaa": "VIII",
            "visible": "1",
            "vscale": "V6",
            "wallchar": "A",
            "walldesc": "Sl채bi",
            "wallid": "770",
            "yds": "5.11d",
          },
        ],
        "soontoberemoved": "0",
        "wallchar": "A",
        "wallname": "Sl채bi",
      },
      "B Suora": Object {
        "problems": Array [
          Object {
            "added": "2018-12-27 14:04:03",
            "addedformatted": "27.12.2018",
            "addedrelative": "2 weeks ago",
            "addt": "",
            "attr_bodytension": "0",
            "attr_complex": "1",
            "attr_compression": "0",
            "attr_coordination": "0",
            "attr_crimpers": "0",
            "attr_dyno": "0",
            "attr_endurance": "0",
            "attr_fingery": "0",
            "attr_hooks": "0",
            "attr_jugs": "0",
            "attr_mantle": "0",
            "attr_pinches": "1",
            "attr_pockets": "0",
            "attr_powerful": "0",
            "attr_sensitive": "1",
            "attr_slopers": "0",
            "attr_technical": "1",
            "attr_volumes": "0",
            "australian": "21",
            "author": "Tuukka",
            "c_dangerous": "0",
            "c_dirty": "0",
            "c_dislike": "0",
            "c_like": "0",
            "c_love": "0",
            "chartcolor": "#26B99A",
            "code": "#40E0D0",
            "color133": null,
            "colour": "Turquoise",
            "colourcode": "#40E0D0",
            "colourid": "38",
            "crag_route_id": null,
            "enddefinition": "Last hold",
            "ext_id": null,
            "font": "6b+",
            "gradeid": "8",
            "gradename": "6B+",
            "gymid": "132",
            "hidedate": null,
            "htmlcode": "#40E0D0",
            "id": "38",
            "lastWall": "Suora",
            "name": "Turquoise",
            "problemid": "47429",
            "routetype": "boulder",
            "score": "550",
            "slider_complexity": "0",
            "slider_intensity": "0",
            "slider_risk": "0",
            "soontoberemoved": "0",
            "sort": null,
            "south_africa": null,
            "stars": null,
            "startdefinition": "Two holds",
            "tag": "BKE2018B034",
            "tagshort": "B034",
            "tapecolour": "#FFFF00",
            "textcolor": "#000",
            "tick": null,
            "uiaa": "VII",
            "visible": "1",
            "vscale": "V4",
            "wallchar": "B",
            "walldesc": "Suora",
            "wallid": "712",
            "yds": "5.11a",
          },
          Object {
            "added": "2018-12-27 14:04:41",
            "addedformatted": "27.12.2018",
            "addedrelative": "2 weeks ago",
            "addt": "",
            "attr_bodytension": "1",
            "attr_complex": "0",
            "attr_compression": "0",
            "attr_coordination": "0",
            "attr_crimpers": "0",
            "attr_dyno": "0",
            "attr_endurance": "0",
            "attr_fingery": "0",
            "attr_hooks": "0",
            "attr_jugs": "0",
            "attr_mantle": "1",
            "attr_pinches": "0",
            "attr_pockets": "0",
            "attr_powerful": "0",
            "attr_sensitive": "1",
            "attr_slopers": "1",
            "attr_technical": "1",
            "attr_volumes": "0",
            "australian": "19",
            "author": "Tuukka",
            "c_dangerous": "0",
            "c_dirty": "0",
            "c_dislike": "0",
            "c_like": "0",
            "c_love": "0",
            "chartcolor": "#BDC3C7",
            "code": "#FF0000",
            "color133": "blue",
            "colour": "Red",
            "colourcode": "#FF0000",
            "colourid": "8",
            "crag_route_id": null,
            "enddefinition": "Last hold",
            "ext_id": null,
            "font": "6a",
            "gradeid": "5",
            "gradename": "6A",
            "gymid": "132",
            "hidedate": null,
            "htmlcode": "#FF0000",
            "id": "8",
            "name": "Red",
            "problemid": "47430",
            "routetype": "boulder",
            "score": "400",
            "slider_complexity": "0",
            "slider_intensity": "0",
            "slider_risk": "0",
            "soontoberemoved": "0",
            "sort": null,
            "south_africa": null,
            "stars": null,
            "startdefinition": "One hold",
            "tag": "BKE2018B035",
            "tagshort": "B035",
            "tapecolour": "#FFFF00",
            "textcolor": "#000",
            "tick": null,
            "uiaa": "VI+",
            "visible": "1",
            "vscale": "V3",
            "wallchar": "B",
            "walldesc": "Suora",
            "wallid": "712",
            "yds": "5.10a",
          },
          Object {
            "added": "2018-12-27 14:31:26",
            "addedformatted": "27.12.2018",
            "addedrelative": "2 weeks ago",
            "addt": "",
            "attr_bodytension": "1",
            "attr_complex": "0",
            "attr_compression": "0",
            "attr_coordination": "1",
            "attr_crimpers": "0",
            "attr_dyno": "0",
            "attr_endurance": "0",
            "attr_fingery": "0",
            "attr_hooks": "0",
            "attr_jugs": "0",
            "attr_mantle": "0",
            "attr_pinches": "0",
            "attr_pockets": "0",
            "attr_powerful": "1",
            "attr_sensitive": "1",
            "attr_slopers": "0",
            "attr_technical": "1",
            "attr_volumes": "0",
            "australian": "25",
            "author": "Tuukka",
            "c_dangerous": "0",
            "c_dirty": "0",
            "c_dislike": "0",
            "c_like": "0",
            "c_love": "0",
            "chartcolor": "#34495E",
            "code": "#FDD017",
            "color133": null,
            "colour": "Yellow",
            "colourcode": "#FDD017",
            "colourid": "4",
            "crag_route_id": null,
            "enddefinition": "Last hold",
            "ext_id": null,
            "font": "7a+",
            "gradeid": "12",
            "gradename": "7A+",
            "gymid": "132",
            "hidedate": null,
            "htmlcode": "#FDD017",
            "id": "4",
            "name": "Yellow",
            "problemid": "47431",
            "routetype": "boulder",
            "score": "750",
            "slider_complexity": "0",
            "slider_intensity": "0",
            "slider_risk": "0",
            "soontoberemoved": "0",
            "sort": null,
            "south_africa": null,
            "stars": null,
            "startdefinition": "Two holds",
            "tag": "BKE2018B036",
            "tagshort": "B036",
            "tapecolour": "#FF0000",
            "textcolor": "#000",
            "tick": null,
            "uiaa": "VIII+",
            "visible": "1",
            "vscale": "V7",
            "wallchar": "B",
            "walldesc": "Suora",
            "wallid": "712",
            "yds": "5.12a",
          },
          Object {
            "added": "2018-12-27 14:43:35",
            "addedformatted": "27.12.2018",
            "addedrelative": "2 weeks ago",
            "addt": "",
            "attr_bodytension": "1",
            "attr_complex": "0",
            "attr_compression": "1",
            "attr_coordination": "1",
            "attr_crimpers": "0",
            "attr_dyno": "0",
            "attr_endurance": "0",
            "attr_fingery": "0",
            "attr_hooks": "0",
            "attr_jugs": "0",
            "attr_mantle": "0",
            "attr_pinches": "0",
            "attr_pockets": "0",
            "attr_powerful": "0",
            "attr_sensitive": "0",
            "attr_slopers": "1",
            "attr_technical": "0",
            "attr_volumes": "1",
            "australian": "22",
            "author": "Antero",
            "c_dangerous": "0",
            "c_dirty": "0",
            "c_dislike": "0",
            "c_like": "1",
            "c_love": "0",
            "chartcolor": "#3498DB",
            "code": "#FF6600",
            "color133": "red",
            "colour": "Orange",
            "colourcode": "#FF6600",
            "colourid": "5",
            "crag_route_id": null,
            "enddefinition": "Last hold",
            "ext_id": null,
            "font": "6c",
            "gradeid": "9",
            "gradename": "6C",
            "gymid": "132",
            "hidedate": null,
            "htmlcode": "#FF6600",
            "id": "5",
            "name": "Orange",
            "problemid": "47433",
            "routetype": "boulder",
            "score": "600",
            "slider_complexity": "0",
            "slider_intensity": "0",
            "slider_risk": "0",
            "soontoberemoved": "0",
            "sort": null,
            "south_africa": null,
            "stars": null,
            "startdefinition": "One hold",
            "tag": "BKE2018B037",
            "tagshort": "B037",
            "tapecolour": "#FFFF00",
            "textcolor": "#000",
            "tick": null,
            "uiaa": "VII+",
            "visible": "1",
            "vscale": "V5",
            "wallchar": "B",
            "walldesc": "Suora",
            "wallid": "712",
            "yds": "5.11b",
          },
        ],
        "soontoberemoved": "0",
        "wallchar": "B",
        "wallname": "Suora",
      },
      "C Kaarikulma": Object {
        "problems": Array [
          Object {
            "added": "2018-12-27 14:45:36",
            "addedformatted": "27.12.2018",
            "addedrelative": "2 weeks ago",
            "addt": "",
            "attr_bodytension": "1",
            "attr_complex": "0",
            "attr_compression": "1",
            "attr_coordination": "0",
            "attr_crimpers": "0",
            "attr_dyno": "0",
            "attr_endurance": "0",
            "attr_fingery": "0",
            "attr_hooks": "0",
            "attr_jugs": "0",
            "attr_mantle": "0",
            "attr_pinches": "0",
            "attr_pockets": "0",
            "attr_powerful": "1",
            "attr_sensitive": "0",
            "attr_slopers": "1",
            "attr_technical": "0",
            "attr_volumes": "1",
            "australian": "20",
            "author": "Antero",
            "c_dangerous": "0",
            "c_dirty": "0",
            "c_dislike": "0",
            "c_like": "0",
            "c_love": "0",
            "chartcolor": "#9B59B6",
            "code": "#00FF00",
            "color133": null,
            "colour": "Neon green",
            "colourcode": "#00FF00",
            "colourid": "33",
            "crag_route_id": null,
            "enddefinition": "Last hold",
            "ext_id": null,
            "font": "6a+",
            "gradeid": "6",
            "gradename": "6A+",
            "gymid": "132",
            "hidedate": null,
            "htmlcode": "#00FF00",
            "id": "33",
            "lastWall": "Kaarikulma",
            "name": "Neon green",
            "problemid": "47435",
            "routetype": "boulder",
            "score": "450",
            "slider_complexity": "0",
            "slider_intensity": "0",
            "slider_risk": "0",
            "soontoberemoved": "0",
            "sort": null,
            "south_africa": null,
            "stars": null,
            "startdefinition": "Two holds",
            "tag": "BKE2018C033",
            "tagshort": "C033",
            "tapecolour": "#FFFF00",
            "textcolor": "#000",
            "tick": null,
            "uiaa": "VII-",
            "visible": "1",
            "vscale": "V3",
            "wallchar": "C",
            "walldesc": "Kaarikulma",
            "wallid": "713",
            "yds": "5.10c",
          },
          Object {
            "added": "2018-12-27 15:07:29",
            "addedformatted": "27.12.2018",
            "addedrelative": "2 weeks ago",
            "addt": "",
            "attr_bodytension": "1",
            "attr_complex": "0",
            "attr_compression": "1",
            "attr_coordination": "0",
            "attr_crimpers": "0",
            "attr_dyno": "0",
            "attr_endurance": "0",
            "attr_fingery": "0",
            "attr_hooks": "0",
            "attr_jugs": "0",
            "attr_mantle": "0",
            "attr_pinches": "0",
            "attr_pockets": "0",
            "attr_powerful": "1",
            "attr_sensitive": "0",
            "attr_slopers": "1",
            "attr_technical": "0",
            "attr_volumes": "0",
            "australian": "20/21",
            "author": "Antero",
            "c_dangerous": "0",
            "c_dirty": "0",
            "c_dislike": "0",
            "c_like": "0",
            "c_love": "0",
            "chartcolor": "#455C73",
            "code": "#FF0066",
            "color133": null,
            "colour": "Pink",
            "colourcode": "#FF0066",
            "colourid": "6",
            "crag_route_id": null,
            "enddefinition": "Last hold",
            "ext_id": null,
            "font": "6b",
            "gradeid": "7",
            "gradename": "6B",
            "gymid": "132",
            "hidedate": null,
            "htmlcode": "#FF0066",
            "id": "6",
            "name": "Pink",
            "problemid": "47436",
            "routetype": "boulder",
            "score": "500",
            "slider_complexity": "0",
            "slider_intensity": "0",
            "slider_risk": "0",
            "soontoberemoved": "0",
            "sort": null,
            "south_africa": null,
            "stars": null,
            "startdefinition": "One hold",
            "tag": "BKE2018C034",
            "tagshort": "C034",
            "tapecolour": "#FFFF00",
            "textcolor": "#000",
            "tick": null,
            "uiaa": "VII",
            "visible": "1",
            "vscale": "V4",
            "wallchar": "C",
            "walldesc": "Kaarikulma",
            "wallid": "713",
            "yds": "5.10d",
          },
        ],
        "soontoberemoved": "0",
        "wallchar": "C",
        "wallname": "Kaarikulma",
      },
    },
    "type": "SET_PROBLEMS",
  },
]
`;

exports[`Testing ProblemList component renders as expected 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
